### PR TITLE
Preserve inherited choice info in `daml test` coverage report

### PR DESCRIPTION
**Pull request by akrmn**
Closes https://github.com/digital-asset/daml/issues/13044

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/lib/DA/Cli/Damlc/Test.hs [hunk 1]
```diff
@@ -177,11 +179,12 @@ printTestCoverage ShowCoverage {getShowCoverage} extPkgs modules results
...
     choices =
-        [ (pidM, m, t, n)
+        [ (pidM, m, t, n, mi)
         | (pidM, m, t) <- templates
-        , n <- concat
-            [ NM.names (LF.tplChoices t)
-            , S.toList . LF.tpiInheritedChoiceNames =<< NM.toList (LF.tplImplements t)
+        , (n, mi) <- concat
+            [ (, Nothing) <$> NM.names (LF.tplChoices t)
+            , NM.toList (LF.tplImplements t) >>= \LF.TemplateImplements {..} ->
```
Comment (robin-da) [inline compiler/damlc/lib/DA/Cli/Damlc/Test.hs:186]: maybe there's a clearer way to write this, I find it fairly hard to read.

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM.

File: compiler/damlc/lib/DA/Cli/Damlc/Test.hs [hunk 2]
```diff
@@ -177,11 +179,18 @@ printTestCoverage ShowCoverage {getShowCoverage} extPkgs modules results
...
     templates = [(pidM, m, t) | (pidM, m) <- modules, t <- NM.toList $ LF.moduleTemplates m]
     choices =
-        [ (pidM, m, t, n)
+        [ (pidM, m, t, n, mi)
         | (pidM, m, t) <- templates
-        , n <- concat
-            [ NM.names (LF.tplChoices t)
-            , S.toList . LF.tpiInheritedChoiceNames =<< NM.toList (LF.tplImplements t)
+        , (n, mi) <- concat
+            [ [ (choice, Nothing)
```
Comment (akrmn) [inline compiler/damlc/lib/DA/Cli/Damlc/Test.hs:185]: @robin-da, do you find this version easier to follow?

Comment (robin-da) [reply compiler/damlc/lib/DA/Cli/Damlc/Test.hs:185]: yes, that looks much better to me.thanks!

**Review by sofiafaro-da — APPROVED** [review]
Looks good, thank you!

**Pull request merged**